client/network/src/protocol/sync.rs

```diff
@@ -1094,9 +1106,9 @@ impl<B: BlockT> ChainSync<B> {
 		if let PeerSyncState::AncestorSearch(_, _) = peer.state {
 			return OnBlockAnnounce::Nothing
 		}
-		// If the announced block is the best they have seen, our common number
+		// If the announced block is the best they have and is not ahead of us, our common number
 		// is either one further ahead or it's the one they just announced, if we know about it.
-		if is_best {
+		if is_best && number > self.best_queued_number {
```

Comment:
Why don't we only update the common number when the number is higher than your best queued number?

Reply:
This is explained in description and the comment above.

> When receiving announcement for some known block we set common block for that peer to be the announced block. The code assumes that if the block is known it is lower than our best blocks. This is not true since reorgs to previous (lower) blocks is possible. Fixed that check.